src/storage/src/hummock/store/version.rs

```diff
@@ -456,32 +415,50 @@ impl HummockReadVersion {
     }
 
     pub fn add_merged_imm(&mut self, merged_imm: ImmutableMemtable) {
-        let staging_imm_count = self.staging.imm.len();
-        let merged_imm_ids = merged_imm.get_imm_ids();
-
-        #[cfg(debug_assertions)]
-        {
-            // check the suffix `merged_imm_ids.len()` imms in staging.imm are the same as
-            // `merged_imm_ids`
-            let diff = staging_imm_count - merged_imm_ids.len();
-            let mut count: usize = 0;
-            for (i, imm) in self.staging.imm.iter().skip(diff).enumerate() {
-                count += 1;
-                assert_eq!(
-                    imm.batch_id(),
-                    merged_imm_ids[i],
-                    "merged_imm_ids: {:?}",
-                    merged_imm_ids
-                );
+        assert!(merged_imm.get_imm_ids().iter().rev().is_sorted());
```

Comment:
This assertion may be too strong and cannot hold, since we don't guarantee imm id is sorted in `staging.imm` and uploader.
```rust
// Note: Currently, building imm and writing to staging version is not atomic, and therefore
// imm of smaller batch id may be added later than one with greater batch id
pub imm: VecDeque<ImmutableMemtable>
```

Reply:
This note is only for handling imms from multiple shards in uploader. For imms in single shard, since they are generated and written in single linear thread, the freshness of data can be directly reflected by comparing batch id, and therefore the batch id should be sorted. Since the input imms in a merging task must be a consecutive subset of imms, they should be sorted in batch id as well. 

In uploader, though the imm ids are not sorted globally, for the subset of imms from a single shard, their imm ids are sorted. A sanity check is added in this PR on monotonically increasing imm id when sealing data and add imm to each shard.

Reply:
Got it, candidate imms to merge for a single shard is sorted in the uploader.